File: Data_Structure/Queue/Implement_Queue_Using_Stack.c
#include<stdio.h>
#include<stdlib.h>
#define MAX 5

int queueStack[MAX];
int top = -1;

//function to remove an element from stack
int pop()
{
    return queueStack[top--];
}

//function to insert an element into stack
void push(int item)
{
    if(top == MAX-1)
    {
        printf("Stack Overflow");
    }
    else
    {
        top++;
        queueStack[top] = item;
    }
}

//function to insert element in the queue using recursive stack call
void enQueue(int item)
{
    push(item);
}

//function to dequeue element from a queue using recursive stack call
int deQueue()
{
    int data, res;
    if(top == -1)
    {
        printf("\nQueue is empty !!");
    }
    else if(top == 0)
    {
        return pop();
    }
    data = pop();
    res = deQueue();
    push(data);
    return res;
}

void peek()
{
    printf("Front element of queue is %d", queueStack[top]);
}

void display()
{
    for(int i= 0; i<=top; i++)
        {
            printf("%d ", queueStack[i]);
        }
        printf("\n\n");
}

int main()
{
    
    enQueue(26);
    enQueue(35);
    enQueue(56);
    enQueue(62);
    
    printf("Queue of elements is:\n");
    display();
    
    peek();
    deQueue();
    
    printf("\n\nQueue of elements is:\n");
    display();
    return 0;
}

File: Data_Structure/DS_Using_Python/Implement_Queue_Using_Stack.py
class Queue:
	def __init__(self):
		self.s1 = Stack()
		self.s2 = Stack()
		
	def isEmpty(self):
		res = self.s1.isEmpty() and self.s2.isEmpty()
		return res
		
	def enQueue(self, item):
		res = self.s1.push(item)
		return res
		
	def deQueue(self):
		if self.s2.isEmpty():
			while not self.s1.isEmpty():
				x = self.s1.pop()
				self.s2.push(x)
		return self.s2.pop()
		
	def display(self):
		res1 = self.s1.stack
		for i in res1:
			print(i, end= ' ')
			
	def show(self):
		res = self.s2.stack
		for i in res[::-1]:
			print(i, end= ' ')
		
class Stack:
	def __init__(self):
		self.stack = []
		
	def isEmpty(self):
		return self.stack == []
		
	def push(self, item):
		self.stack.append(item)
		
	def pop(self):
		return self.stack.pop()
		
q = Queue()
q.enQueue(4)
q.enQueue(6)
q.enQueue(10)
q.enQueue(15)
q.enQueue(19)

print("Queue of elements is:")
q.display()

result = q.deQueue()
print("\n\nDeleted item is:", result)

print("\nAfter removing an element:")
q.show()